src/features/todos.ts

```diff
@@ -1,8 +1,41 @@
 import { createSlice } from '@reduxjs/toolkit';
 import { Todo } from '../types/Todo';
 
+interface TodosState {
+  items: Todo[];
+  status: 'idle' | 'loading' | 'succeeded' | 'failed';
+}
+
+const initialState: TodosState = {
+  items: [],
+  status: 'idle',
+};
+
 export const todosSlice = createSlice({
   name: 'todos',
-  initialState: [] as Todo[],
-  reducers: {},
+  initialState,
+  reducers: {
+    setTodos(state, action) {
+      // eslint-disable-next-line no-param-reassign
+      state.items = action.payload;
+      // eslint-disable-next-line no-param-reassign
+    },
+    setLoading(state) {
+      // eslint-disable-next-line no-param-reassign
+      state.status = 'loading';
+    },
```

Comment:
The `setLoading` reducer seems to be responsible for setting the loading state. However, it would be more appropriate to have this functionality within async thunks, where you can set the loading state before starting the async operation and reset it once the operation is complete. This way, you can avoid having to manually call `setLoading` and `setEndLoading` every time you perform an async operation.